x-pack/plugins/infra/server/lib/metrics/lib/calculate_bucket_size/calculate_auto.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import moment, { isDuration, Duration } from 'moment';
+import { isNumber } from 'lodash';
+const d = moment.duration;
+
+const roundingRules = [
+  [d(500, 'ms'), d(100, 'ms')],
+  [d(5, 'second'), d(1, 'second')],
+  [d(7.5, 'second'), d(5, 'second')],
+  [d(15, 'second'), d(10, 'second')],
+  [d(45, 'second'), d(30, 'second')],
+  [d(3, 'minute'), d(1, 'minute')],
+  [d(9, 'minute'), d(5, 'minute')],
+  [d(20, 'minute'), d(10, 'minute')],
+  [d(45, 'minute'), d(30, 'minute')],
+  [d(2, 'hour'), d(1, 'hour')],
+  [d(6, 'hour'), d(3, 'hour')],
+  [d(24, 'hour'), d(12, 'hour')],
+  [d(1, 'week'), d(1, 'd')],
+  [d(3, 'week'), d(1, 'week')],
+  [d(1, 'year'), d(1, 'month')],
+  [Infinity, d(1, 'year')],
+];
+
+const revRoundingRules = roundingRules.slice(0).reverse();
+
+type NumberOrDuration = number | Duration;
+
+type Rule = NumberOrDuration[];
+
+type CheckFunction = (
+  bound: NumberOrDuration,
+  interval: Duration,
+  target: number
+) => Duration | undefined;
+
+function find(rules: Rule[], check: CheckFunction, last?: boolean) {
+  function pick(buckets: number, duration: Duration) {
```

Comment:
`find` and `pick` are methods I'm expecting to be imported from Lodash. Can you use more specific names on these functions?